lldbinit.py

```diff
@@ -9,16 +9,174 @@
 import re
 import os
 
-# Bazel builds all binaries from a username-specific cache location
-# for lldb to work properly we need to map that path back to our main source
-project_root = os.path.dirname(os.path.realpath(__file__))
-bazel_workspace = os.path.realpath(os.path.join(project_root, "bazel-sorbet"))
-
-if not(os.path.isdir(bazel_workspace)):
-    print("You need to compile Sorbet first before loading this file")
-else:
-    print("Mapping %s to current working directory" % bazel_workspace)
-    source_map_command = 'settings set -- target.source-map "%s" "%s"' % (bazel_workspace, project_root)
-    ci = lldb.debugger.GetCommandInterpreter()
-    res = lldb.SBCommandReturnObject()
-    ci.HandleCommand(source_map_command, res)
+def __lldb_init_module(debugger, internal_dict):
+    result = lldb.SBCommandReturnObject()
+    init_source_map(debugger, result)
+    ci = debugger.GetCommandInterpreter()
+    ci.HandleCommand("command script add -f lldbinit.cmd_dump_sorbet dumpsorbet", result)
+    register_summary(ci, format_sorbet_core_Name.__name__, "sorbet::core::Name")
+    register_summary(ci, format_sorbet_core_NameRef.__name__, "sorbet::core::NameRef")
+    register_summary(ci, format_sorbet_core_GlobalState.__name__, "sorbet::core::GlobalState")
+    # LLDB will not automatically walk up inheritance chains to find a formatter so it needs to
+    # be specified for every subclass of Instruction
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Instruction")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Ident")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::Send")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::CFG")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::VariableUseSite")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::cfg::LocalRef")
+    register_summary(ci, format_sorbet_object.__name__, "sorbet::core::FileRef")
+
+def register_summary(ci, funcName, typeName):
+    result = lldb.SBCommandReturnObject()
+    ci.HandleCommand("type summary add -F %s.%s %s" % (__name__, funcName, typeName), result)
+    return result
+
+def init_source_map(debugger, result):
+    # Bazel builds all binaries from a username-specific cache location
+    # for lldb to work properly we need to map that path back to our main source
+    project_root = os.path.dirname(os.path.realpath(__file__))
+    bazel_workspace = os.path.realpath(os.path.join(project_root, "bazel-sorbet"))
+
+    if not os.path.isdir(bazel_workspace):
+        result.Print("You need to compile Sorbet first before loading this file")
+    else:
+        result.Print("Mapping %s to current working directory" % bazel_workspace)
+        source_map_command = 'settings set -- target.source-map "%s" "%s"' % (bazel_workspace, project_root)
```

Comment:
Not all files are in `bazel-sorbet`. in particular, files built by `genrule`'s are in subfolders of `bazel-out`.

Reply:
What do you mean? Generated files don't need to be mapped back to the main source tree so why would it matter? (as in lldb should be using the bazel-out path there).